src/retrieve-validation/index.ts

```diff
@@ -20,13 +21,45 @@ export interface W3CCSSValidatorResponse {
 	};
 }
 
+// Helper function that checks if a function has been called with the correct types and throws an error otherwise
+const validateRetrievalCall = async (
+	retrievalFunction: typeof retrieveValidation,
+	method: 'GET' | 'POST',
+	parameters: W3CValidatorParameters | string,
+	timeout: number
+): ReturnType<typeof retrieveValidation> => {
+	if (method === 'GET' && typeof parameters === 'string') {
+		return await retrievalFunction(method, parameters, timeout);
+	}
+
+	if (method === 'POST' && typeof parameters === 'object') {
+		return await retrievalFunction(method, parameters, timeout);
+	}
+
+	throw new Error('Validation function called with unsupported parameters');
+};
```

Comment:
The validation here is directly for Typescript, the overload definitions don't allow for the implementation definition, (string is more generic than `GET | POST`). I can include `string` as a type for method in the overload definitions, which is what I initially did, but I built it this way to make it more strict. If I type it more generally, Typescript will indicate that passing any string to the function will result in the same return value as calling it with `POST | GET`, but I could maybe throw an error at the top of the function to prevent such usage. Its not as big of a concern since like you said this function isn't exposed, but keeping the types strict is good for maintainability down the line.